src/icmp/Icmp4.cc

```diff
@@ -173,21 +173,19 @@ Icmp4::Recv(void)
     if (pkt == nullptr)
         pkt = (char *)xmalloc(MAX_PKT4_SZ);
 
-    Ip::Address::InitAddr(from);
     n = recvfrom(icmp_sock,
                  (void *)pkt,
                  MAX_PKT4_SZ,
                  0,
-                 from->ai_addr,
-                 &from->ai_addrlen);
+                 reinterpret_cast<sockaddr*>(&from),
+                 0);
```

Comment:
The horrible casting aside, this change violates recvfrom() call intent (and possibly [API](https://linux.die.net/man/2/recvfrom)) by supplying a nil pointer to address size (disguised as integer `0`): "the caller should initialize [addrlen memory] before the call to the size of the buffer associated with src_addr"

To move this PR forward, I suggest:

* Test changes before posting a PR (or PR updates).
* Remove reinterpret_cast. If you need a sockaddr object, use a local sockaddr variable .
* Read recvfrom() documentation and understand why that specific system call is used here.
* Supply a non-nil addrlen pointer to recvfrom(). Use another local variable of the right type for this addrlen. The value of that variable will depend on how your solution provides the memory for the address, but it will not be zero.
* Add a check for recvfrom() truncating the address, report an error if it did, and do _not_ use the truncated address.
* Do not use std::move() unless the code does not compile _without_ it (its use is always a red flag in Squid code, but there are legitimate use cases, of course): It is difficult to use std::move() correctly, it is usually a bad idea to use it, and it is probably not going to be needed in this PR.